### ISSUE TITLE
Syntax highlighting broken for JSON blocks

### ISSUE THREAD

**Issue by domenjesenovec** [hashicorp/vscode-terraform#476]
After one of the recent upgrades (most likely 2.2.0 or 2.2.1) the syntax highlighting got broken in one of my tf files. The code that breaks highlighting:
```javascript
locals {
  dockerconfigjson1 = {
    "auths" : {
      "docker.io" = {
        email    = "user@company.com"
        username = "user"
        password = "password567"
        auth     = base64encode(join(":", ["user", "password567"]))
      }
    }
  }
  dockerconfigjson2 = {
    "auths" : {
      "company-docker-repo.net:6555" = {
        email    = "user@company.com"
        username = "user"
        password = "password123"
        auth     = base64encode(join(":", ["user", "password123"]))
      }
    }
  }
}
```

After the end of dockerconfigjson1 block the highlighting is broken - all the text is highlighted in the same color.

**Comment by vibhuyadav** [comment]
I can report the same thing happening.
![Screen Shot 2020-08-25 at 6 24 31 PM](https://user-images.githubusercontent.com/6370789/91233935-49a83c80-e700-11ea-98dc-3dcf63a77568.png)

**Comment by aeschright** [comment]
Thanks for the lengthy examples @domenjesenovec and @vibhuyadav – I can see where the problem is so I'm prioritizing another patch release to fix this. Stay tuned! In the meantime, you may want to downgrade to an earlier version.

**Comment by vibhuyadav** [comment]
What's the piece that's breaking? Just curious

**Comment by aeschright** [comment]
It's to do with the syntax matcher needing to be flexible enough to catch all of the possible things that might be assigned to those object keys, but also not treat the closing bracket as part of the value – basically I fixed the regular expression that matches the brackets by tightening up what could be assigned as a value, but now it's failing to match your function calls.

**Comment by aeschright** [comment]
The update for this has been released in v2.2.2.
